test/test-core.js

```diff
@@ -247,6 +247,159 @@
 		});
 	});
 
+	t('Dynamic response status callback - list of statuses as an array (errors only)', function(assert) {
```

Comment:
These tests aren't actually looping... you only call `$.ajax()` once in each, or am I missing something? In any case, since you've split out the various tests into successful status codes and error status codes I think you're fine. We can just check that the returned status code is in the "possible" ones and be done with this. So no need for the third test, and we can simplify the other two. Let me know if you have questions!